### PR TITLE
Issue #731 MaxExpansionDepth of 0 is not work

### PR DESCRIPTION
**Pull request by VikingsFan**
### Issues
This pull request fixes issue #731 .

### CODE REVIEW

**Comment by xuzhg** [comment]
@luoyan0517 is added to the review. #Closed

File: OData/test/E2ETest/WebStack.QA.Test.OData/DollarLevels/DollarLevelsTest.cs [hunk 1]
```diff
@@ -194,5 +194,30 @@ public static void UpdateConfiguration(HttpConfiguration configuration)
             Assert.Contains(errorMessage,
                 result);
         }
+
+        [Theory]
```
Comment (luoyan0517) [inline OData/test/E2ETest/WebStack.QA.Test.OData/DollarLevels/DollarLevelsTest.cs]: Add a case with 2 $levels in a query, like $expand=Manager($levels=max;$expand=DirectReports($levels=2))?

**Comment by luoyan0517** [comment]
:shipit:

**Comment by foobardog** [comment]
Sorry it took me a while to respond.

This will definitely stop the 500 exception, but then MaxExpansionDepth of 0 acts a little weird. Basically you'll get:

$levels=1 -> 1 level
$levels=2 -> 2 levels
$levels=3 -> 3 levels
...
$levels=99 -> 99 levels
$levels=max -> 2 levels.

This puts it at odds with the [OData Spec](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part2-url-conventions/odata-v4.0-errata02-os-part2-url-conventions-complete.html#_Toc406398162):

> The value of the $levels option is either a positive integer to specify the number of levels to expand, or the literal string max to specify the maximum expansion level supported by that service.

I personally can be fine with that assuming the documentation around MaxExpansionDepth of 0 can be updated.

**Comment by VikingsFan** [comment]
@foobardog if $level=max, it need a value to limit, or it's a dead loop, customer can set with MaxDepth or it should be default value 2.

**Comment by foobardog** [comment]
Right, I'm fine with this fix as long as the documentation gets updated to make it clear what MaxExpansionDepth 0 will mean.

:shipit:

**Comment by VikingsFan** [comment]
Merged 47e33ca0873f61ac37539ce4860943dead865959, Thanks!

**Comment by xuzhg** [comment]
@VikingsFan Please update the related document.

**Comment by VikingsFan** [comment]
Doc: be21510d0e828398b979b875ac2ae2d30bdd63f6

**Pull request closed without merging**